sklearn/preprocessing/_discretization.py

```diff
@@ -272,7 +341,11 @@ def fit(self, X, y=None, sample_weight=None):
             elif self.strategy == "quantile":
                 quantiles = np.linspace(0, 100, n_bins[jj] + 1)
                 if sample_weight is None:
-                    bin_edges[jj] = np.asarray(np.percentile(column, quantiles))
+
+                    bin_edges[jj] = np.asarray(
+                        np.percentile(column, quantiles, method=self.quantile_method),
+                        dtype=np.float64,
+                    )
                 else:
                     bin_edges[jj] = np.asarray(
                         [
```

Comment:
The following line should also be updated to implement a symmetrized version of the call to `_weighted_percentile` when `quantile_method="average_inverted_cdf"`.

For now, we can implement this as a function named `_symmetrized_weighted_percentile` in the same file as the latter along with a new unit tests that check the expected weighted / repetition properties are preserved when symmetrizing and the equivalence with `"average_inversted_cdf"` method of NumPy when using unit weights.